src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -2008,6 +2036,7 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
             IMethodSymbol debugEntryPoint = null,
             Stream sourceLinkStream = null,
             IEnumerable<EmbeddedText> embeddedTexts = null,
+            Stream metadataPeStream = null,
```

Comment:
I'll capitalize.
As for the name, ref/contract assemblies are only one kind of metadata-only assemblies. That's why I used this name.